pkg/repository/udmrepo/kopialib/backend/common_kopia_algorithms_test.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/kopia/kopia/repo/encryption"
+	"github.com/kopia/kopia/repo/format"
+	"github.com/kopia/kopia/repo/hashing"
+	"github.com/kopia/kopia/repo/splitter"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/vmware-tanzu/velero/pkg/repository/udmrepo"
+)
+
+func TestSetupNewRepoAlgorithms(t *testing.T) {
+
+	testCases := []struct {
+		name     string
+		envVars  map[string]string
+		flags    map[string]string
+		expected repo.NewRepositoryOptions
+	}{
+		{
+			name: "with valid non-default hash algo from env",
+			envVars: map[string]string{
+				"KOPIA_HASHING_ALGORITHM": "HMAC-SHA3-224",
+			},
+			flags: map[string]string{},
+			expected: repo.NewRepositoryOptions{
+				BlockFormat: format.ContentFormat{
+					Hash:       "HMAC-SHA3-224",
+					Encryption: encryption.DefaultAlgorithm,
+				},
+				ObjectFormat: format.ObjectFormat{
+					Splitter: splitter.DefaultAlgorithm,
+				},
+			},
+		},
+		{
+			name: "with valid non-default encryption algo from env",
+			envVars: map[string]string{
+				"KOPIA_HASHING_ALGORITHM":    "",
+				"KOPIA_SPLITTER_ALGORITHM":   "",
+				"KOPIA_ENCRYPTION_ALGORITHM": "CHACHA20-POLY1305-HMAC-SHA256",
+			},
+			flags: map[string]string{},
+			expected: repo.NewRepositoryOptions{
+				BlockFormat: format.ContentFormat{
+					Hash:       hashing.DefaultAlgorithm,
+					Encryption: "CHACHA20-POLY1305-HMAC-SHA256",
+				},
+				ObjectFormat: format.ObjectFormat{
+					Splitter: splitter.DefaultAlgorithm,
+				},
+			},
+		},
+		{
+			name:    "with valid non-default splitter algo from env",
+			envVars: map[string]string{"KOPIA_SPLITTER_ALGORITHM": "FIXED-512K"},
+			flags:   map[string]string{},
+			expected: repo.NewRepositoryOptions{
+				BlockFormat: format.ContentFormat{
+					Hash:       hashing.DefaultAlgorithm,
+					Encryption: encryption.DefaultAlgorithm,
+				},
+				ObjectFormat: format.ObjectFormat{
+					Splitter: "FIXED-512K",
+				},
+			},
+		},
+		{
+			name:    "with valid non-default splitter and hashing algo from env, invalid encryption from env",
+			envVars: map[string]string{"KOPIA_SPLITTER_ALGORITHM": "FIXED-512K", "KOPIA_HASHING_ALGORITHM": "HMAC-SHA3-224", "KOPIA_ENCRYPTION_ALGORITHM": "NON-EXISTING-SHA256"},
+			flags:   map[string]string{},
+			expected: repo.NewRepositoryOptions{
+				BlockFormat: format.ContentFormat{
+					Hash:       "HMAC-SHA3-224",
+					Encryption: encryption.DefaultAlgorithm,
+				},
+				ObjectFormat: format.ObjectFormat{
+					Splitter: "FIXED-512K",
+				},
+			},
+		},
+		{
+			name:    "with unsupported hash algo in env, fallback to default",
+			envVars: map[string]string{"KOPIA_HASHING_ALGORITHM": "unsupported-hash"},
+			flags:   map[string]string{},
+			expected: repo.NewRepositoryOptions{
+				BlockFormat: format.ContentFormat{
+					Hash:       hashing.DefaultAlgorithm,
+					Encryption: encryption.DefaultAlgorithm,
+				},
+				ObjectFormat: format.ObjectFormat{
+					Splitter: splitter.DefaultAlgorithm,
+				},
+			},
+		},
+		{
```

Comment:
@mpryc Should we add another testcase when `StoreOptionGenHashAlgos` has a valid algo value but the env var has an invalid algo value. Then in such a scenario default would not be used but the one present in `StoreOptionGenHashAlgos` would be used, right ?

Reply:
@shubham-pampattiwar right. there are 2 levels of fallback here:
1) use env var if it's set and value is supported, if not, fall back to...
2) value in StoreOptionGenHashAlgo, if not fall back to...
3) default